lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -34,30 +49,50 @@
 
 
 @attr('shard_1')
+@ddt.ddt
 class AccessTestCase(LoginEnrollmentTestCase, ModuleStoreTestCase):
     """
     Tests for the various access controls on the student dashboard
     """
+    TOMORROW = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)
+    YESTERDAY = datetime.datetime.now(pytz.utc) - datetime.timedelta(days=1)
+
     def setUp(self):
         super(AccessTestCase, self).setUp()
         course_key = SlashSeparatedCourseKey('edX', 'toy', '2012_Fall')
         self.course = course_key.make_usage_key('course', course_key.run)
         self.anonymous_user = AnonymousUserFactory()
+        self.beta_user = BetaTesterFactory(course_key=self.course.course_key)
         self.student = UserFactory()
         self.global_staff = UserFactory(is_staff=True)
         self.course_staff = StaffFactory(course_key=self.course.course_key)
         self.course_instructor = InstructorFactory(course_key=self.course.course_key)
+        self.staff = GlobalStaffFactory()
 
-    def verify_access(self, mock_unit, student_should_have_access):
+    def verify_access(self, mock_unit, student_should_have_access, expected_error_type=None):
         """ Verify the expected result from _has_access_descriptor """
         self.assertEqual(
             student_should_have_access,
-            access._has_access_descriptor(self.anonymous_user, 'load', mock_unit, course_key=self.course.course_key)
+            bool(access._has_access_descriptor(
+                self.anonymous_user, 'load', mock_unit, course_key=self.course.course_key))
         )
+
         self.assertTrue(
             access._has_access_descriptor(self.course_staff, 'load', mock_unit, course_key=self.course.course_key)
         )
 
+        if expected_error_type is not None:
+            self.assertIsInstance(
+                access._has_access_descriptor(self.anonymous_user, 'load', mock_unit,
+                                              course_key=self.course.course_key),
+                expected_error_type)
```

Comment:
Let's refactor this method.  It's a bit confusing:
First, check anonymous user's access, comparing booleans
Then, check staff user's access
Then, check anonymous user's access again, comparing error types